cpp/src/parquet/column_writer.h

```diff
@@ -87,8 +88,9 @@ class PARQUET_EXPORT PageWriter {
 
   static std::unique_ptr<PageWriter> Open(
       std::shared_ptr<ArrowOutputStream> sink, Compression::type codec,
-      int compression_level, ColumnChunkMetaDataBuilder* metadata,
```

Comment:
Let's not remove this API but create a new one taking `const CodecOptions&`, to avoid breaking API compatibility.

Reply:
Should we mark the old one as deprecated? The new one should always be preferable and we don't want to maintain different overloads in the long term.

Reply:
We can indeed.